### PR TITLE
Feature/BA-859 - Approval Digitization

### PR DESCRIPTION
**Pull request by sutherlanda**
feat(opportunities): Create workflow for ADM approval of opportunities prior to publication

- Add in Approval tab on CWU opportunity edit view
- Approval form containing request info and contract info
- Approval and contract data stored on opportunity data model
- Message/email templates for approval requests and notifications for approval actioning
- 2FA via SMS or email for secure approval

Fixes BA-859.

### CODE REVIEW

File: config/lib/seed.js [hunk 1]
```diff
@@ -1,381 +1,406 @@
...
 }
 
-function seedTestMessageTemplate () {
+function seedTestMessageTemplate() {
 	console.log(chalk.yellow('Database seeding:\tSeeding message templates.'));
-	var T = mongoose.model ('MessageTemplate');
-	var saveT = function (t) {
-		return new Promise (function (resolve, reject) {
-			t.save (function (err) {
+	var T = mongoose.model('MessageTemplate');
```
Comment (patricksimonian) [inline config/lib/seed.js]: I'd suggest replace `T` with `MessageTemplate` just as a convention since other Mongoose model declarations do that

Comment (SteveChapmanBCDX) [reply config/lib/seed.js]: Good suggestion, but we won't do that refactoring right now.

File: config/message-templates/approval-request.html [hunk 2]
```diff
@@ -0,0 +1,506 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta name="viewport" content="width=device-width" />
+  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+  <title>Email for Final Approver</title>
+  <style>
```
[the remark this answers is not in the document]
Comment (SteveChapmanBCDX) [reply config/message-templates/approval-request.html]: Agreed.

File: config/message-templates/pre-approval-request.html [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+
+      .btn-primary a:hover {
+        background-color: #34495e !important;
+        border-color: #34495e !important;
+      }
+    }
+  </style>
+</head>
+
+<body class="">
```
[the remark this answers is not in the document]
Comment (sutherlanda) [reply config/message-templates/pre-approval-request.html]: Agreed.  @mark-a-wilson can you take a look at the two templates you provided for this feature and clean up the CSS and ideally, abstract it out into a separate file like @patricksimonian has suggested?

File: modules/core/client/directives/forms.client.directive.js [hunk 4]
```diff
@@ -148,6 +130,9 @@
 			if (this.options.hasOwnProperty('disabled')) {
```
[the remark this answers is not in the document]
Comment (sutherlanda) [reply modules/core/client/directives/forms.client.directive.js]: Will change, but not in this PR.

File: modules/opportunities/client/controllers/opportunities.client.controller.js [hunk 5]
```diff
@@ -139,50 +201,14 @@
...
-					'May',
-					'June',
-					'July',
-					'August',
-					'September',
-					'October',
-					'November',
-					'December'
-				];
+				var monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
```
[the remark this answers is not in the document]
Comment (sutherlanda) [reply modules/opportunities/client/controllers/opportunities.client.controller.js]: Yes, that is definitely something we should do.  Will leave that for a separate PR.

File: config/message-templates/pre-approval-request.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (mark-a-wilson) [reply config/message-templates/pre-approval-request.html]: ok!

[the remark this answers is not in the document]
Comment (mark-a-wilson) [reply config/message-templates/pre-approval-request.html]: 8182753

**Comment by sutherlanda** [comment]
@SteveChapmanBCDX, @mark-a-wilson this feature is ready for final review.  When you are satisfied, I will merge into dev as discussed.

**Pull request merged**